react/providers/api/provider.tsx

```diff
@@ -2,29 +2,45 @@ import React, { useEffect, useReducer } from 'react'
 import { useChains } from '../../hooks/mod.ts'
 import { APIContext } from './context.ts'
 import { apiProvidersReducer } from './reducer.ts'
-import { useConfig } from '../../mod.ts'
-import { ApiPromise, WsProvider } from "../../../core/mod.ts";
+import { ProviderConnection, useConfig } from '../../mod.ts'
+import { ApiPromise, Sc, ScProvider, WsProvider } from "../../../core/mod.ts";
+import { ChainId } from '../../../chains/types.ts'
+import { ProviderConnectionConfig } from '../config/mod.ts'
+
+const isTrusted = (chainId: ChainId, apiConfig?: ProviderConnectionConfig): boolean => {
+  if (!apiConfig) return true;
+  if (apiConfig?.[chainId] === 'light-client') return false;
+  if (!apiConfig?.[chainId] && apiConfig.default === 'light-client') return false;
+
+  return true
+}
 
 export const APIProvider: React.FC<React.PropsWithChildren<any>> = ({
   children,
 }) => {
   const chains = useChains()
-  const { chainRpcs } = useConfig()
+  const c = useConfig()
   const [apis, dispatch] = useReducer(apiProvidersReducer, {})
 
   useEffect(() => {
     chains.forEach((chain) => {
-      const provider = new WsProvider(chainRpcs[chain.id] || chain.rpcs[0])
+      const connection: ProviderConnection = isTrusted(chain.id, c.api) ? ({
```

Comment:
Here we only create one type of provider per chain. I'm not sure if this will work for light clients if they take time to init, but this is what I was thinking originally